File: app/src/main/java/team/singularity/scoutapp2022/Teams.java
package team.singularity.scoutapp2022;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.TextView;
import android.widget.AdapterView.OnItemClickListener;
import androidx.appcompat.app.AppCompatActivity;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;
import java.util.List;
import java.util.Map;
import kotlin.Metadata;
import kotlin.collections.IntIterator;
import kotlin.jvm.internal.Intrinsics;
import kotlin.ranges.RangesKt;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class Teams extends AppCompatActivity {
    private final String tag = "Teams";
    private ListView view;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_teams);

        view = findViewById(R.id.teams_layout);

        view.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view1, int i, long l) {
                // start MatchInformation page
                Log.i(Teams.this.tag, "Clicked teams view");
            }
        });
    }

    public void onResume() {
        super.onResume();

        List list = (List)(new ArrayList());

        SimpleAdapter adapter = new SimpleAdapter(this,
                list,
                R.layout.teams_layout,
                new String[]{"name", "number"},
                new int[]{R.id.teamName, R.id.teamNumber});

        view.setAdapter((ListAdapter)adapter);

        view.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                String teamNumber = ((TextView)findViewById(R.id.teamNumber)).getText().toString();
                Intent match = new Intent(Teams.this.getApplicationContext(), MatchActivity.class);
                match.putExtra("Team Number", teamNumber);
                startActivity(match);
                finish();
            }
        });

        // bluetooth code

        // iterate over all teams in the database
        for (int i = 0; i < DatabaseClass.teamData.length(); i++) {

            // set name and number
            HashMap resultsMap = new HashMap();
            ((Map)resultsMap).put("name", DatabaseClass.getTeamName(i));
            ((Map)resultsMap).put("number", String.valueOf(DatabaseClass.getTeamNumber(i)));
            list.add(resultsMap);
        }

        // iterate over local teams
        /*
        (0 until Database.tempTeamData().length()).forEach {
            val resultsMap = HashMap<String, String>()

            resultsMap["name"] = getLocalTeamName(i)
            resultsMap["number"] = "${getLocalTeamNumber(i)}"
            list.add(resultsMap)
        }
         */
    }
}

File: app/src/main/java/team/singularity/scoutapp2022/Teams.kt
package team.singularity.scoutapp2022

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.ListView
import android.widget.SimpleAdapter
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity


class Teams : AppCompatActivity() {
    private val tag = "Teams"
    private lateinit var view: ListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_teams)

        this.view = findViewById(R.id.teams_layout)
        this.view.setOnItemClickListener { adapterView: AdapterView<*>, view1: View, i: Int, l: Long ->
            // start MatchInformation page
            Log.i(tag, "Clicked teams view")
        }
    }

    override fun onResume() {
        super.onResume()

        val list = mutableListOf<HashMap<String, String>>()

        val adapter = SimpleAdapter(this,
            list,
            R.layout.teams_layout,
            arrayOf("name", "number"),
            intArrayOf(R.id.teamName, R.id.teamNumber))

        view.adapter = adapter

        view.setOnItemClickListener { adapter, view, i, l ->
            /* TODO : make this more direct */
            val teamNumber = (view.findViewById<View>(R.id.teamNumber) as TextView).text.toString()
            val match = Intent(applicationContext, Match::class.java)
            match.putExtra("Team Number", teamNumber)
            startActivity(match)
            finish()
        }

        // bluetooth code

        // iterate over all teams in the database
        // for (int i = 0; i < teamData.length(); i++)
        for (i in 0 until Database.teamData().length()) {
            val resultsMap = HashMap<String, String>()

            // set name and number
            resultsMap["name"] = getTeamName(i)
            resultsMap["number"] = "${getTeamNumber(i)}"
            list.add(resultsMap)
        }

        // iterate over local teams
        for (i in 0 until Database.tempTeamData.length()) {
            val resultsMap = HashMap<String, String>()

            resultsMap["name"] = getLocalTeamName(i)
            resultsMap["number"] = "${getLocalTeamNumber(i)}"
            list.add(resultsMap)
        }
    }
}
